src/libs/histogram.c

```diff
@@ -354,19 +391,31 @@ static void _lib_histogram_vectorscope_bkgd(dt_lib_histogram_t *d, const dt_iop_
       dt_aligned_pixel_t rgb_scope, XYZ_D50, chromaticity;
       for_each_channel(ch, aligned(vertex_rgb, delta, rgb_scope:16))
         rgb_scope[ch] = vertex_rgb[k][ch] + delta[ch] * i;
-      dt_ioppr_rgb_matrix_to_xyz(rgb_scope, XYZ_D50, vs_prof->matrix_in_transposed, vs_prof->lut_in,
-                                 vs_prof->unbounded_coeffs_in, vs_prof->lutsize, vs_prof->nonlinearlut);
-      if(vs_type == DT_LIB_HISTOGRAM_VECTORSCOPE_CIELUV)
+      if(vs_type != DT_LIB_HISTOGRAM_VECTORSCOPE_RYB)
```

Comment:
see comment below where I think a swicth() should be easier to read and have a clear separate entries for the 3 supported cases.